lib/interpreters/split_clustering.rb

```diff
@@ -129,6 +120,302 @@ def self.generate_split_vrps(service_vrp, _job = nil, block = nil)
+      empties_or_fills = vrp.services.select{ |s| s.quantities.any?(&:fill) || s.quantities.any?(&:empty) }
+      vrp.services -= empties_or_fills
+      # TODO: 0- relations needs to be taken into account inside clustering during this split
+      #          - the services need to be in the same route (and by extension, in the same sub-vrp) for the following relations:
+      #            => same_route, order, sequence, shipment
+      split_by_vehicle = split_balanced_kmeans(service_vrp, vrp.vehicles.size, cut_symbol: :duration, restarts: 2, build_sub_vrps: false)
+
+      # ss_data
+      service_vrp[:split_level] = 0
+      service_vrp[:split_solve_data] = {
+        current_vehicles: vrp.vehicles.map{ |v| v }, # new array but original objects
+        current_vehicle_limit: vrp.resolution_vehicle_limit,
+        transferred_empties_or_fills: empties_or_fills.map{ |s| s }, # new array but original objects
+        transferred_vehicles: [],
+        transferred_vehicle_limit: vrp.resolution_vehicle_limit && 0,
+        transferred_time_limit: 0.0,
+        service_vehicle_assignments: vrp.vehicles.map.with_index{ |v, i| [v[:id], split_by_vehicle[i]] }.to_h,
+        original_vrp: vrp,
+        split_vrp: nil,
+      }
+      transfer_empty_vehicles, service_vrp[:split_solve_data][:current_vehicles] =
+        service_vrp[:split_solve_data][:current_vehicles].partition{ |vehicle|
+          service_vrp[:split_solve_data][:service_vehicle_assignments][vehicle.id].empty?
+        }
+      service_vrp[:split_solve_data][:transferred_vehicles].concat(transfer_empty_vehicles)
+      transfer_empty_vehicles.each{ |v| service_vrp[:split_solve_data][:service_vehicle_assignments].delete(v.id) }
+      service_vrp[:split_solve_data][:split_vrp] = create_fake_split_vrp(service_vrp[:split_solve_data])
+
+      # Then split the services into left-and-right groups by
+      # using the service_vehicle_assignments information
+      # (don't generate any sub-VRP yet)
+      new_split_solve_core(service_vrp, job, &block) # self-recursive method
+    ensure
+      service_vrp[:vrp] = service_vrp[:split_solve_data][:original_vrp]
+      service_vrp[:vrp].services.concat empties_or_fills
+      log '<-- new_split_solve (clustering by max_split)'
+    end
+
+    # self-recursive method
+    def self.new_split_solve_core(service_vrp, job = nil, &block)
+      split_level = service_vrp[:split_level]
+      ss_data = service_vrp[:split_solve_data]
+
+      if split_solve_candidate?(service_vrp) # if it still needs splitting
+        enum_current_vehicles = ss_data[:current_vehicles].select # Enumerator to keep a local copy of current_vehicles
+        current_vehicle_limit = ss_data[:current_vehicle_limit]
+
+        # SPLIT current_vehicles list (by-vehicle-centroids) to create two "sides"
+        sides = split_balanced_kmeans(
+          { vrp: create_fake_sub_split_vrp(ss_data) }, 2,
```

Comment:
Using the work 'fake' seems like (to me) we are creating some stub object and not really representative objects

Reply:
I quite agree with this, it seems it is a temporary object which does not matter.
What about ```create_current_sub_split_vrp```?

Reply:
I renamed it to `representative_sub_vrp` as we decided